rskj-core/src/main/java/co/rsk/mine/MinerServerImpl.java

```diff
@@ -377,7 +383,15 @@ public MinerWork updateGetWork(@Nonnull final Block block, @Nonnull final boolea
     }
 
     public void setExtraData(byte[] extraData) {
-        this.extraData = extraData;
+        RLPList decodedExtraData = RLP.decodeList(this.extraData);
+        byte[] version = decodedExtraData.get(0).getRLPData();
+        byte[] identity = decodedExtraData.get(1).getRLPData();
+        this.extraData = RLP.encodeList(version, RLP.encode(identity), RLP.encodeElement(extraData));
```

Comment:
If the above modification is to be done, this should be modified to something like:
```
        RLPList decodedExtraData = RLP.decodeList(this.extraData);
        byte[] version = decodedExtraData.get(0).getRLPData();
        byte[] identity = decodedExtraData.get(1).getRLPData();
        this.extraData = RLP.encodeList(version, identity, RLP.encodeElement(extraData));
```

Reply:
From what I see, the only difference is `RLP.encode(identity)` v `identity`. 
`RLP.encode(identity)` is needed for having a correct encoding.
Let me know if I'm missing something!